bindings.go

```diff
@@ -59,25 +70,43 @@ type Input struct {
 	Encoding InputEncoding
 }
 
+var ErrOperationLimit = errors.New("operation limit was hit")
+var ErrNoLanguage = errors.New("cannot parse without language")
+
 // Parse produces new Tree from content using old tree
-func (p *Parser) Parse(oldTree *Tree, content []byte) *Tree {
+func (p *Parser) Parse(ctx context.Context, oldTree *Tree, content []byte) (*Tree, error) {
 	var BaseTree *C.TSTree
 	if oldTree != nil {
 		BaseTree = oldTree.c
 	}
 
+	parseComplete := make(chan struct{}, 1)
```

Comment:
**Suggestion:** Implement this as an unbuffered channel and just use closing the channel as the signal.